crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -427,7 +427,10 @@ impl<'db> TypeInferenceBuilder<'db> {
                 );
             }
             DefinitionKind::ExceptHandler(handler) => {
-                self.infer_except_handler_definition(handler, definition);
+                self.infer_except_handler_definition(handler, definition, false);
+            }
+            DefinitionKind::ExceptStarHandler(handler) => {
+                self.infer_except_handler_definition(handler, definition, true);
```

Comment:
I think if we are going to just collapse back down to a boolean flag on the inference end, then we should just pass it through the definition as a boolean flag also.

(I know this is opposite to what I was leaning toward before, but that was on the assumption that the inference end would be more separated. But it's all the same node, so I think that was wrong.)

Reply:
Yeah, it worked out differently to how I was expecting as well... I'll switch it to be just a boolean flag!